src/core/net/dns_headers.hpp

```diff
@@ -493,6 +493,136 @@ class Name
         kMaxEncodedLength = 255, ///< Max length of an encoded name.
     };
 
+    /**
+     * This enumeration represent the name type.
+     *
+     */
+    enum Type : uint8_t
+    {
+        kTypeEmpty,   ///< The name is empty (not specified).
+        kTypeCString, ///< The name is given as a C string (dot '.' separated sequence of labels').
+        kTypeMessage, ///< The name is specified from a message at a given offset (encoded in the message).
+    };
+
+    /**
+     * This constructor initializes the `Name` object as empty (not specified).
+     *
+     */
+    Name(void)
+        : Name(nullptr, nullptr, 0)
+    {
+    }
+
+    /**
+     * This constructor initializes the `Name` object with a given string.
+     *
+     * @param[in] aString   A string specifying the name (dot '.' separated sequence of labels').
+     *
+     */
+    explicit Name(const char *aString)
+        : Name(aString, nullptr, 0)
+    {
+    }
+
+    /**
+     * This constructor initializes the `Name` object from a message at a given offset.
+     *
+     * @param[in] aMessage   The message containing the encoded name. `aMessage.GetOffset()` MUST point to the start of
+     *                       the DNS header in the message (used to parse compressed name).
+     * @param[in] aOffset    The offset in @p aMessage pointing to the start of the name.
+     *
+     */
+    Name(const Message &aMessage, uint16_t aOffset)
+        : Name(nullptr, &aMessage, aOffset)
+    {
+    }
+
+    /**
+     * This method indicates whether the name is empty (not specified).
+     *
+     * @returns TRUE if the name is empty, FALSE otherwise.
+     *
+     */
+    bool IsEmpty(void) const { return (mString == nullptr) && (mMessage == nullptr); }
+
+    /**
+     * This method indicates whether the name is specified from a C string.
+     *
+     * @returns TRUE if the name is specified from a string, FALSE otherwise.
+     *
+     */
+    bool IsFromCString(void) const { return mString != nullptr; }
+
+    /**
+     * This method indicates whether the name is specified from a message.
+     *
+     * @returns TRUE if the name is specified from a message, FALSE otherwise.
+     *
+     */
+    bool IsFromMessage(void) const { return mMessage != nullptr; }
+
+    /**
+     * This method gets the type of `Name` object indicating whether it is empty, specified by a C string or from a
+     * message
+     *
+     * @returns The name type.
+     *
+     */
+    Type GetType(void) const { return IsFromCString() ? kTypeCString : (IsFromMessage() ? kTypeMessage : kTypeEmpty); }
```

Comment:
nit: should we be more specific about what kind of type?
```suggestion
    Type GetFromType(void) const { return IsFromCString() ? kTypeCString : (IsFromMessage() ? kTypeMessage : kTypeEmpty); }
```